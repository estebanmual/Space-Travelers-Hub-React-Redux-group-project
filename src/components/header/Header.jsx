import React from 'react';
import logo from '../../images/planet.svg';
import style from './Header.module.css';

function Header() {
  return (
    <nav className={style.Header}>
      <div className={style.HeaderLeft}>
        <img className={style.HeaderLogo} src={logo} alt="Space Travelers Hub logo" />
        <h1 className={style.HeaderTitle}>Space Travelers&apos; Hub</h1>
      </div>
      <div />
    </nav>
  );
}

export default Header;
