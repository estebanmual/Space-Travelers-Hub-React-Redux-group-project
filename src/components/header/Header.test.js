import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from './Header';

describe('testing Header', () => {
  it('snapshot for the header ', () => {
    const tree = renderer.create(<Router><Header /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('testing Header component renders correctly', () => {
    render(<Router><Header /></Router>);
    expect(screen.getByText('My Profile')).toBeInTheDocument();
  });
});
