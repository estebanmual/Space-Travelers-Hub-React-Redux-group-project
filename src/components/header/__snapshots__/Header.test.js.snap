// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Header snapshot for the header  1`] = `
<nav
  className="Header"
>
  <div
    className="HeaderLeft"
  >
    <img
      alt="Space Travelers Hub logo"
      className="HeaderLogo"
      src="planet.svg"
    />
    <h1
      className="HeaderTitle"
    >
      Space Travelers' Hub
    </h1>
  </div>
  <div
    className="HeaderRight"
  >
    <a
      aria-current="page"
      className="aTag active"
      href="/"
      onClick={[Function]}
    >
      Rockets
    </a>
    <a
      className="aTag"
      href="/Missions"
      onClick={[Function]}
    >
      Missions
    </a>
    <a
      className="aTag"
      href="/MyProfile"
      onClick={[Function]}
    >
      My Profile
    </a>
  </div>
</nav>
`;
